packages/api-docs/src/generatePageJsonFiles.ts

```diff
@@ -23,7 +26,24 @@ export function generatePageJsonFiles(options: IPageJsonOptions): void {
   const apiModel = new ApiModel();
   for (const apiJsonPath of apiJsonPaths) {
     console.log('Loading ' + apiJsonPath);
-    apiModel.loadPackage(apiJsonPath);
+
+    // If the file belongs to the compat layer.
+    if (apiJsonPath.indexOf('compat') !== -1) {
+      // Get the json info and change the package name.
+      const jsonObject = require(apiJsonPath);
+      jsonObject.name = jsonObject.name + '-compat';
+
+      // Write the info into a temporary file.
+      fs.writeFileSync(tmpFile, JSON.stringify(jsonObject));
```

Comment:
Nit, it's better to use fs-extra (`fse`) here--it has all the same methods plus some improvements, and is already used in this file. Also it has a convenience method `writeJSONSync` which is nice here.